File: src/assignment/_11To20/P17_AreaOfSpace.java
package assignment._11To20;
/*
 * program to find the area of space between two concentric circles
 */

import java.util.Scanner;

import static java.lang.System.out;

// class declared as package private and final to prevent inheritance
final class P17_AreaOfSpace {
    // initializing instance for scanner class as constant
    private static final Scanner INPUT = new Scanner(System.in);

    // method to get double input from the user
    private static double getUserInput() throws Exception{
        if(!INPUT.hasNextDouble()){
            throw new Exception("Invalid input...");
        }else{
            return INPUT.nextDouble();
        }
    }

    // method to calculate the area of the cylinder
    private double findAreaOfCircle(double radius){
        return Math.PI * radius * radius;
    }

    public static void main(String [] args){
        P17_AreaOfSpace program17 = new P17_AreaOfSpace();
        try{
            out.println("\nTool to find the area of space between two concentric circles ");
            out.print("Enter the radius of the first Circle : ");
            double radius1 = getUserInput();                  // user input of first circle radius

            out.print("Enter the radius of the second Circle : ");
            double radius2 = getUserInput();                  // user input of second circle radius

            double area1 = program17.findAreaOfCircle(radius1);         // area of circle1
            double area2 = program17.findAreaOfCircle(radius2);         // area of circle2
            out.printf("\nArea of space between two circles of radius %.2f and radius %.2f is %.2f\n", radius1, radius2, Math.abs(area1-area2));
        }catch (Exception e){
            out.println(e.getMessage());
        }
    }
}

File: src/kotlin_assignment/_11To20/P17_AreaOfSpace.kt
package kotlin_assignment._11To20;

import kotlin_assignment.getDouble
import kotlin.math.abs

/*
 * program to find the area of space between two concentric circles
 */


// method to calculate the area of circle
fun calculateTheAreaOfCircleWith(radius:Double): Double{
    return 2 * Math.PI * radius * radius;
}

// method to calculate the area space between two concentric circles
fun areaOfSpaceBetween(area1:Double, area2:Double):String{
    return String.format("%.2f", abs(area1- area2));
}
fun main(){
    println("Tool to calculate the area of space between two concentric circle ")
    val radius1 :Double  = getDouble("Enter the radius of the circle 1 : ");
    val radius2 :Double = getDouble("Enter the radius of the 2nd circle : ");
    val area1 = calculateTheAreaOfCircleWith(radius1); // calculating the radius of the circle with radius1
    var area2 = calculateTheAreaOfCircleWith(radius2); // calculating the radius of the circle with radius2
    print("Area of space between the two concentric circle with radius $radius1 and $radius2 is : ${areaOfSpaceBetween(area1, area2)} sq.units");
}